### DISCUSSION TITLE
Using GitHub Copilot for MLOps, TDD, & more

### DISCUSSION THREAD

**Discussion by mikethemelanson** [community/community#57344] [Copilot Conversations]
### Select Topic Area

General

### Body

![The ReadME Project logo](https://user-images.githubusercontent.com/8250055/101071884-0efd3e00-3552-11eb-8523-11305d3cc848.png)
***
Hey folks! :wave:

@mikethemelanson, senior editor with GitHub’s [The ReadME Project](https://github.com/readme), here. 

We have three [guides](https://github.com/readme/guides?src=rmpcod) in our May edition, contributed by developers like you, on the different ways they use generative AI such as GitHub Copilot for software development. 

### [Harness the power of generative AI for software development](https://github.com/readme/guides/coding-generative-ai?src=rmpcod)
Software developer Anton Mirhorodchenko (@GreenWizard2015) explains how he combined GitHub Copilot with ChatGPT to design and build software, from architecture to code:
>I recently conducted research on the effective use of AI for creating quality code, which I published in an open source project called [User Country Analytics](https://github.com/GreenWizard2015/User-Country-Analytics/). Instead of focusing solely on the ability of AI to solve problems, I wanted to determine AI’s ability to use best practices, including testing and other aspects. [...] Just like in real life, I started the project by consulting with an expert: a [software architect](https://github.com/GreenWizard2015/User-Country-Analytics/blob/main/AI%20Backlog/00%20-%20Software%20Architect%20Expert.md). In this case, however, the expert was not human, but rather ChatGPT playing the role of software architect.

### [The case for using Rust in MLOps](https://github.com/readme/guides/rust-mlops?src=rmpcod)
You might be used to using Python, but [Pragmatic Labs](https://github.com/paiml) founder Noah Gift (@noahgift) explains that combining [GitHub Codespaces](https://github.com/features/codespaces) with GitHub Copilot can help you get started quickly using a more performant language like Rust:
>The hard choice between the performant but complex C++ and the easy-to-learn but comparatively slow Python ultimately results in many companies choosing Python. But there is another way. Rust consistently ranks among the [most performant and energy-efficient languages](https://greenlab.di.uminho.pt/wp-content/uploads/2017/10/sleFinal.pdf). It also happens to be among the most loved languages in [StackOverflow’s annual developer survey](https://survey.stackoverflow.co/2022/). 

### [Accelerate test-driven development (TDD) with AI](https://github.com/readme/guides/github-copilot-automattic?src=rmpcod)
TDD can deliver better code, but it also means writing more of it…so let GitHub Copilot handle that part and get there quicker, says Gio Lodi (@mokagio), a mobile infrastructure engineer at [Automattic](https://github.com/Automattic):
>In a real-world codebase, that kind of contrived implementation would get noticed early, but it's a way for me to introduce a helpful testing technique called triangulation. When you don't trust a test's ability to detect false positives, add more tests. GitHub Copilot makes triangulating tests a breeze because generating new code based on previously written code is precisely what the AI was trained to do.

:question: **What ways are you using GitHub Copilot?** Is it all about that boilerplate? Is Copilot making unfamiliar APIs easy to access? Or are you jumping into new languages like @mokagio? :question:

Tell us what you’re working on!

***
**The ReadME Project** is a GitHub platform dedicated to highlighting the best from the open source software community—the people and tech behind projects you use every day. To get the latest stories, articles, and best-practices from The ReadME Project delivered directly to your inbox, [subscribe to The ReadME Project newsletter](https://github.com/readme/#newsletter?src=rmpcod).

**Comment by MuhammadazizEsonboyev** [comment]
mezza

**Comment by melikaaaaaa** [comment]
Mezza

**Comment by Lab-Top** [comment]
> ### Select Topic Area
> 
> General
> 
> ### Body
> 
> ![The ReadME Project logo](https://user-images.githubusercontent.com/8250055/101071884-0efd3e00-3552-11eb-8523-11305d3cc848.png)
> ***
> Hey folks! :wave:
> 
> @mikethemelanson, senior editor with GitHub’s [The ReadME Project](https://github.com/readme), here. 
> 
> We have three [guides](https://github.com/readme/guides?src=rmpcod) in our May edition, contributed by developers like you, on the different ways they use generative AI such as GitHub Copilot for software development. 
> 
> ### [Harness the power of generative AI for software development](https://github.com/readme/guides/coding-generative-ai?src=rmpcod)
> Software developer Anton Mirhorodchenko (@GreenWizard2015) explains how he combined GitHub Copilot with ChatGPT to design and build software, from architecture to code:
> >I recently conducted research on the effective use of AI for creating quality code, which I published in an open source project called [User Country Analytics](https://github.com/GreenWizard2015/User-Country-Analytics/). Instead of focusing solely on the ability of AI to solve problems, I wanted to determine AI’s ability to use best practices, including testing and other aspects. [...] Just like in real life, I started the project by consulting with an expert: a [software architect](https://github.com/GreenWizard2015/User-Country-Analytics/blob/main/AI%20Backlog/00%20-%20Software%20Architect%20Expert.md). In this case, however, the expert was not human, but rather ChatGPT playing the role of software architect.
> 
> ### [The case for using Rust in MLOps](https://github.com/readme/guides/rust-mlops?src=rmpcod)
> You might be used to using Python, but [Pragmatic Labs](https://github.com/paiml) founder Noah Gift (@noahgift) explains that combining [GitHub Codespaces](https://github.com/features/codespaces) with GitHub Copilot can help you get started quickly using a more performant language like Rust:
> >The hard choice between the performant but complex C++ and the easy-to-learn but comparatively slow Python ultimately results in many companies choosing Python. But there is another way. Rust consistently ranks among the [most performant and energy-efficient languages](https://greenlab.di.uminho.pt/wp-content/uploads/2017/10/sleFinal.pdf). It also happens to be among the most loved languages in [StackOverflow’s annual developer survey](https://survey.stackoverflow.co/2022/). 
> 
> ### [Accelerate test-driven development (TDD) with AI](https://github.com/readme/guides/github-copilot-automattic?src=rmpcod)
> TDD can deliver better code, but it also means writing more of it…so let GitHub Copilot handle that part and get there quicker, says Gio Lodi (@mokagio), a mobile infrastructure engineer at [Automattic](https://github.com/Automattic):
> >In a real-world codebase, that kind of contrived implementation would get noticed early, but it's a way for me to introduce a helpful testing technique called triangulation. When you don't trust a test's ability to detect false positives, add more tests. GitHub Copilot makes triangulating tests a breeze because generating new code based on previously written code is precisely what the AI was trained to do.
> 
> :question: **What ways are you using GitHub Copilot?** Is it all about that boilerplate? Is Copilot making unfamiliar APIs easy to access? Or are you jumping into new languages like @mokagio? :question:
> 
> Tell us what you’re working on!
> 
> ***
> **The ReadME Project** is a GitHub platform dedicated to highlighting the best from the open source software community—the people and tech behind projects you use every day. To get the latest stories, articles, and best-practices from The ReadME Project delivered directly to your inbox, [subscribe to The ReadME Project newsletter](https://github.com/readme/#newsletter?src=rmpcod).
>

**Reply by sardoregamberdiyev** [reply]
hi

**Comment by wasim-abbas** [comment]
thank you for giving me access to Git Hub.
